Source/RunActivity/Viewer3D/Popups/TrainDrivingWindow.cs

```diff
@@ -0,0 +1,1214 @@
+            {Viewer.Catalog.GetString("Steam usage"), Viewer.Catalog.GetString("STEM")},
+            {Viewer.Catalog.GetString("Throttle"), Viewer.Catalog.GetString("THRO")},
+            {Viewer.Catalog.GetString("Time"), Viewer.Catalog.GetString("TIME")},
+            {Viewer.Catalog.GetString("Traction cut-off relay"), Viewer.Catalog.GetString("TRAC")},
+            {Viewer.Catalog.GetString("Train brake"), Viewer.Catalog.GetString("BTRN")},
+            {Viewer.Catalog.GetString("Wheel"), Viewer.Catalog.GetString("WHEL")}
+        };
+
+        private static readonly Dictionary<string, string> LastColToAbbreviated = new Dictionary<string, string>()
+        {
+            {Viewer.Catalog.GetString("apply Service"), Viewer.Catalog.GetString("Apply")},
+            {Viewer.Catalog.GetString("Apply Quick"), Viewer.Catalog.GetString("ApplQ")},
+            {Viewer.Catalog.GetString("Apply Slow"), Viewer.Catalog.GetString("ApplS")},
+            {Viewer.Catalog.GetString("coal"), Viewer.Catalog.GetString("c")},
+            {Viewer.Catalog.GetString("Emergency Braking Push Button"), Viewer.Catalog.GetString("EmerBPB")},
+            {Viewer.Catalog.GetString("Lap Self"), Viewer.Catalog.GetString("LapS")},
+            {Viewer.Catalog.GetString("Minimum Reduction"), Viewer.Catalog.GetString("MRedc")},
+            {Viewer.Catalog.GetString("safe range"), Viewer.Catalog.GetString("safe")},
+            {Viewer.Catalog.GetString("skid"), Viewer.Catalog.GetString("Skid")},
+            {Viewer.Catalog.GetString("slip warning"), Viewer.Catalog.GetString("Warning")},
+            {Viewer.Catalog.GetString("slip"), Viewer.Catalog.GetString("Slip")},
+            {Viewer.Catalog.GetString("water"), Viewer.Catalog.GetString("w")},
+        };
+
+        bool ctrlAIFiremanOn = false; //AIFireman On
+        bool ctrlAIFiremanOff = false;//AIFireman Off
+        bool ctrlAIFiremanReset = false;//AIFireman Reset
+        double clockAIFireTime; //AIFireman reset timing
+
+        bool grateLabelVisible = false;// Grate label visible
+        double clockGrateTime; // Grate hide timing
+
+        bool wheelLabelVisible = false;// Wheel label visible
+        double clockWheelTime; // Wheel hide timing
+
+        bool doorsLabelVisible = false; // Doors label visible
+        double clockDoorsTime; // Doors hide timing
+
+        bool ResizeWindow = false;
+        bool UpdateDataEnded = false;
+        const int TextSize = 15;
+        int FirstColLenght = 0;
+        int FirstColOverFlow = 0;
+        int LastColLenght = 0;
+        int LastColOverFlow = 0;
+        int LinesCount = 0;
+        int WindowHeightMin = 0;
+        int WindowHeightMax = 0;
+        int WindowWidthMin = 0;
+        int WindowWidthMax = 0;
+
+        Label indicator;
+        LabelMono indicatorMono;
+        Label ExpandWindow;
+        Label LabelFontToBold;
+
+        public bool normalTextMode = true;// Standard text
```

Comment:
We now have the ability to save options like this to the settings store so they will be preserved across games, by the way. [See the Track Monitor code.](https://github.com/openrails/openrails/blob/5b55e3fb5fd00412323cc61b73f014084615f277/Source/RunActivity/Viewer3D/Popups/TrackMonitorWindow.cs#L193)

Reply:
Please, could you extend this suggestion?
Does this replace the `protected internal override void Save(BinaryWriter outf)` and the `protected internal override void Restore(BinaryReader inf)` procedures?.